packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -169,25 +166,26 @@ export class ProjectService {
         });
         this.projectLoading[projectUuid] = false;
         this.projectAdapters[projectUuid] = adapter;
-        this.projectModel.saveExploresToCache(projectUuid, explores);
     }
 
     private static async testProjectAdapter(
         data: UpdateProject,
-    ): Promise<[ProjectAdapter, (Explore | ExploreError)[]]> {
+    ): Promise<ProjectAdapter> {
         const adapter = await projectAdapterFromConfig(
             data.dbtConnection,
             data.warehouseConnection,
+            {
+                warehouseCatalog: undefined,
+                onWarehouseCatalogChange: () => {},
+            },
```

Comment:
Isn't this a problem on update a project ? it tests the adaptor with no cache and saves no cache either 🤔  we might need to pass this object as an arg of `testProjectAdapter`.

Reply:
Just been making a decision here with update, do we want to recompile the project or leave that to the user?

Reply:
@ZeRego